src/common.cpp

```diff
@@ -1888,39 +1886,83 @@ std::string get_executable_path(const char *argv0) {
         return std::string(buff);
     }
 #elif defined(__OpenBSD__)
+    int cntp = 0;
+    struct stat st;
+    std::string path;
     bool ok = false;
-    std::string narrow;
+    char exe[buff_size];
+    kinfo_file *kif = nullptr;
     static kvm_t *kd = nullptr;
-    wcstring wargv0 = str2wcstring(argv0);
-    // not sure if I am doing this right??
-    null_environment_t vars = null_environment_t();
-    auto path = path_get_path(wargv0, vars);
-    if (path) {
-        int cntp = 0;
-        struct stat st;
-        kinfo_file *kif = nullptr;
-        char errbuf[_POSIX2_LINE_MAX];
-        narrow = wcs2string(*path); 
-        kd = kvm_openfiles(nullptr, nullptr, nullptr, KVM_NO_FILES, errbuf); 
-        if (kd) {
-            if ((kif = kvm_getfiles(kd, KERN_FILE_BYPID, getpid(), sizeof(struct kinfo_file), &cntp))) {
-                for (int i = 0; i < cntp; i++) {
-                    if (kif[i].fd_fd == KERN_FILE_TEXT) {
-                        if (!stat(narrow.c_str(), &st)) {
-                            if (st.st_dev == (dev_t)kif[i].va_fsid || st.st_ino == (ino_t)kif[i].va_fileid) {
-                                ok = true;
-                                break;
-                            }
-                        }
+    char errbuf[_POSIX2_LINE_MAX];
+    const char *pwd = nullptr, *cwd = nullptr, *penv = nullptr;
+    if (!std::string(argv0).empty()) {
+        if (argv0[0] == '/') {
+            path = argv0;
+            goto finish1;
```

Comment:
Please don't use goto.

Instead, keep the path as an std::string - you're already converting it to and fro twice in many cases.